Doc/c-api/list.rst

```diff
@@ -127,6 +155,12 @@ List Objects
    Return ``0`` on success, ``-1`` on failure.  Indexing from the end of the
    list is not supported.
 
+   .. note::
+
+      If *itemlist* is ``NULL`` to delete items and the finalizer
+      (:meth:`object.__del__`) of a deleted item modify the list, it is
+      possible that the list is longer than expected.
```

Comment:
Same question: is it useful to the reader?

Reply:
If you ask the question, you consider that it's not useful, so I removed the note.